### PR TITLE
516 variable tau

### LINKED ISSUE #516: variable tau value in sia380 model for thermal latency
Currently, we have tau constants from sia 2024 that implicitly represent thermal latency of a construction. Let us make it variable based on construction heat capacities and surface areas!

https://miro.com/app/board/o9J_leSAuA0=/

### LINKED ISSUE #594: sia380 - Natural ventilation - Use actual window opening dimensions
Natural ventilation through rectangular openings can be formulated as

#567 will only use the sqrt of the window area to estimate H and W and keep scope small, so would like to improve this by providing actual dimensions into sia380 to fit the equation for ventilation in rectangular openings.

### LINKED ISSUE #622: Significant change of U values in the building properties don't change the energy demand significantly
Hi 

Again, I am not sure whether it is a bug or not. 

In _exercise 1 - Refurbishment_ - I set under the _building properties_ "Bestand" (**default setting Uwall= 0.8, U window=1.5**) and changed its U values afterwards to very bad values (**Uwall=1.5, U window=2.5**). It changed the annual energy demand only by 4% which I had the impression that could not be right:

**Bestand default setting Uwall= 0.8, U window=1.5**
Bildschirmfoto_Hive_best_defaultsettings 2

**Bestand, changed envelope: Uwall=1.5, U window=2.5**
Bildschirmfoto_Hive_best_Uwa1 5_Uwi2 5

### LINKED ISSUE #623: Hive building input form not large enough in Envelope tab
image

Hi,

as shown in the screenshot, the dialog for defining the building properties is not big enough to accommodate all the information, and I was not able to expand the window nor scroll down. 

Besides, it was not very intuitive that a double-click on the component gives this dialog box, without reading an additional documentation (like the one we receive with the exercises), so it would be nice if such instruction is also in the GH template (like a scribble text or something)

Keyword: ENHANCEMENT

### LINKED ISSUE #644: Hizard - Checing adaptive comfort and natural ventilation does not persist upon saving / reopening
**Describe the bug**
In Hizard, checking and unchecking the adaptive comfort and natural ventilation boxes persists during the grasshopper session but does not write / save it to the file.

**To Reproduce**
1. Go to hizard
2. Enable adaptive comfort and disable natural ventilation
3. Save and close grasshopper file
4. Open file and see both checkboxes are at default again

**Expected behavior**
Keeps checkboxes set as they were set upon save.

**Screenshots**

**Software**
 - Rhino 6.34
 - Hive 0.12 (beta, see #642)

**Additional context**

### PR DESCRIPTION
**Pull request by maxenceryan**
# 516 variable tau

## Issues

Closes #516 

Closes #622 ? seems to work when i change standardwert, zielwert and bestand (example EFH had about 15kwh, 31kwh, 80kwh, which seesm reasonable) as well as when I alter only uvalues in hizard.
Closes #644 ? Somehow got fixed along the way but not sure why. Worth checking this
Closes #594 ? This was a quick fix. Summed all window areas instead of just using one (whcih is fine based on the equation we used, no need to treat each window separately). Also, as per issue (see comments at the end), changed c_d to 0.25 instead of 0.6 as this is correct for single large openings.

(Chris) closes #623  ... at least on my computer?

## Description

- Calculates tau based on surface areas, types, and capacitance per floor area instead of reading it from SIA 2024
- Capacitance per floor area is derived from SIA2024 and propagated to the capacities of other compoennts (wall, window, etc)
- Construction type taken from 4 SIA 380 templates (superlightweight, lightweight, mediumweight, heavyweight) to determine capacitance. User can only select templates ( the rest is generated)
- U-values for sia380 demand calc are per component type rather than opaque / transparent.  Distributor adjusted accordingly

## Checklist

- [x] Remove debugging artifacts (if needed)
- [x] Rebuild the Setup_Hive.exe
- [x] Update Hive Wiki (if needed)
- [x] Update most important Grasshopper template(s); as of now [/GrasshopperExamples/LectureExercises/EaCS3_E04_Hive_Template.gh](https://github.com/architecture-building-systems/hive/blob/master/GrasshopperExamples/LectureExercises/EaCS3_E04_Hive_Template.gh)

### CODE REVIEW

**Comment by maxenceryan** [comment]
progress:
- drop down for construction type loads properly from sia380 records and fills up capacitance per floor area correctly
- roof / wall / floor capacitances in the background are updating correctly on the zone

todo:
- changes in construction type / room capacitance are not persisiting upon close of hizard

File: src/Hive.IO/Hive.IO/GhParametricInputs/GhSIARoom.cs [hunk 1]
```diff
@@ -72,6 +73,7 @@ protected override void SolveInstance(IGH_DataAccess DA)
 
             DA.GetData(0, ref siaRoom.RoomType);
             DA.GetData(1, ref siaRoom.RoomConstant);
+            DA.GetData(1, ref siaRoom.CapacitancePerFloorArea);
```
Comment (maxenceryan) [inline src/Hive.IO/Hive.IO/GhParametricInputs/GhSIARoom.cs:76]: correct this index

File: src/Hive.IO/Hive.IO/Building/Zone.cs [hunk 2]
```diff
@@ -113,14 +113,134 @@ public struct StructInternalLoads
...
         public double WindowArea => Windows.Sum(w => w.Area);
-        public double FloorArea => Floors.Sum(f => f.Area);
+        public double FloorsArea => Floors.Sum(f => f.Area);
 
         #endregion
 
 
+        #region Construction
+
+        public string ConstructionType;
```
Comment (maxenceryan) [inline src/Hive.IO/Hive.IO/Building/Zone.cs:126]: add comment to describe wth this is?

File: src/Hive.IO/Hive.IO/GhDistributors/GhDistSIA.cs [hunk 3]
```diff
@@ -63,14 +61,15 @@ protected override void SolveInstance(IGH_DataAccess DA)
                 {
                     // TO DO: make check that it's not a void
                     zoneAreas[i] += floor.Area;
+                    // TODO no floor on surfaces?
```
Comment (maxenceryan) [inline src/Hive.IO/Hive.IO/GhDistributors/GhDistSIA.cs:64]: check this

Comment (maxenceryan) [reply src/Hive.IO/Hive.IO/GhDistributors/GhDistSIA.cs:64]: seems there was a decision (from the comments) that only surfaces exposed to the environment (not ground) should be part of the surface areas/types passed to sia380. Is this based on the SIA 380 calculation or some other design decision?

Comment (christophwaibel) [reply src/Hive.IO/Hive.IO/GhDistributors/GhDistSIA.cs:64]: yeah that was for determining which surfaces are relevant for transmission heat losses (Q_T). Surfaces to ground are not according to sia380. But if we had an overhang/cantilever and a room above, that floor surface would be actually exposed to the environment and thus relevant for Q_T. OK I have an idea... lemme try sth

Comment (christophwaibel) [reply src/Hive.IO/Hive.IO/GhDistributors/GhDistSIA.cs]: image

these geometries should now work. overhangs & cantilevers will be interpreted as floor area that accounts to

**Review by christophwaibel — APPROVED** [review]
did some testing with the grasshopper files. all features worked as explained: persisting checkboxes, thermal mass has an impact on demand, so does u-values of individual component classes (floors, roofs, windows, walls), adaptive comfort works, natural ventilation works.

I **did not** go through the code because of time, but I guess, since the functionality is given, it's fiiine ;-)

**Pull request merged**